From 4da8c2c9a401c10a41e533d75b3153fe1561b05b Mon Sep 17 00:00:00 2001
From: Ryan Roden-Corrent <ryan@rcorre.net>
Date: Thu, 12 Mar 2020 18:52:44 -0400
Subject: [PATCH] Improve error message for ViewportTexture.

The "ViewportTexture: Path to node is invalid" error can sometimes be
fixed just by ordering the viewport before the node using the texture.

Relates to #27790.
---
 scene/main/viewport.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scene/main/viewport.cpp b/scene/main/viewport.cpp
index c11b11bc71f..fb5c5e41ad3 100644
--- a/scene/main/viewport.cpp
+++ b/scene/main/viewport.cpp
@@ -67,7 +67,7 @@ void ViewportTexture::setup_local_to_scene() {
 	}
 
 	Node *vpn = local_scene->get_node(path);
-	ERR_FAIL_COND_MSG(!vpn, "ViewportTexture: Path to node is invalid.");
+	ERR_FAIL_COND_MSG(!vpn, "ViewportTexture: Path to node is invalid. If the path seems valid, move the node using the texture after the viewport.");
 
 	vp = Object::cast_to<Viewport>(vpn);
 

